groupassigner.php

```diff
@@ -0,0 +1,92 @@
+<?php
+ob_start();
+$groupPage = true;
+require_once __DIR__ . "/include/header.php";
+require_once __DIR__ . "/include/tsutils.php";
+require_once __DIR__ . "/include/cacheutils.class.php";
+
+$cacheutils = new CacheUtils('grouplist');
+
+if($cacheutils->isExpired()) {
+    $cacheutils->setValue([getServerGroups(), date('d.m.Y H:i:s')], 300);
+}
+
+$grouplist = $cacheutils->getValue();
+try {
+    $tsAdmin = getTeamspeakConnection("#no_query_clients");
+} catch (Exception $e){
+    if ($e->getCode() == 1281) {
+        echo '';
+    } else {
+        echo '<div class="alert alert-danger"><p class="text-center">' . translate($lang["general"]["scripterror"], [$e->getCode(), $e->getMessage()]) . '</p></div>';
+        exit;
+    }
+}
+
+
+if(isset($_POST["absenden"])){
+    $uid = htmlspecialchars($_POST['uid']);
+    if(in_array(htmlspecialchars($_POST['group']), $config["groups"]["allowgroups"])){
+        try {
+            $tsAdmin->clientGetByUid($uid)->addServerGroup(htmlspecialchars($_POST['group']));
+            setCookie("tsuid", $uid);
+            $success = "success";
+        } catch(Exception $e) {
+            if(strpos($e, "invalid clientID") == true){
+                $error = "offline";
+            }elseif(strpos($e, "duplicate entry") == true){
+                $error = "duplicate";
+            }
+
+        }
+    }else{
+         $error = "notallowed";
+    }
+}
+?>
+<?php if($config["groupassigner"] == 1){ ?>
```

Comment:
No need to close then open again, just remove 
`?> <?php`

Reply:
I edited it :-)